### ISSUE TITLE
Possible bug in search with multiple stars in path.

### ISSUE THREAD

**Issue by pvlpenev** [akesterson/dpath-python#67]
Consider the following input:
```python
testdata = {
    "a": [
        {
            "b": [
                {"c": 1},
                {"c": 2},
                {"c": 3}
            ]
        }
    ]
}

testpath = "a/*/b/*/c"
```

I would expect `search(testdata, testpath)` to produce the following result(return the same object as the input):

```python
{'a': [{'b': [{'c': 1}, {'c': 2}, {'c': 3}]}]}
```

Instead the result is:
```python
{'a': [{'b': [{'c': 1}]},
       {'b': [{'c': 2}]},
       {'b': [{'c': 3}]}]}
```

Am I misunderstanding the behavior of search?

**Comment by poftwaresatent** [comment]
Maybe related is the following, where search seems to just hang forever:

```
foo=dict(bar=1, baz=list('abc'))
print dpath.util.search(foo, '**')
```

I would have expected that to just print foo, i.e.:

```
{'baz': ['a', 'b', 'c'], 'bar': 1}
```

Interestingly, if you Ctrl-C the search and then print foo, it comes back with a dict that has a very very very long list of repeated `['a', 'b', 'c', ...]` for baz.

**Comment by akesterson** [comment]
@calebcase  has a fix for the 2.0 branch, but this bug persists in 1.x.

**Comment by calebcase** [comment]
Fixed in 2.x and 2.x is now released. Please upgrade and let us know how it goes. Thanks for the report!
